vassal-app/src/main/java/VASSAL/counters/PlaceMarker.java

```diff
@@ -236,6 +236,21 @@ protected Command placeMarker() {
       c = m.placeAt(marker, p);
     }
 
+    // Mat support
+    if ((c != null) && GameModule.getGameModule().isMatSupport()) {
+      // If a cargo piece has been placed, find it a Mat if eligible, and select it if the Mat is selected
+      if ("true".equals(marker.getProperty(MatCargo.IS_CARGO))) { //NON-NLS
```

Comment:
Why's this a `String` instead of a `Boolean`?

Reply:
Possibly I just don't know everything about how getProperty things work, I'm used to the inside-VASSAL end of it where they're all strings. So you're saying this would return me a `Boolean`? Or are you saying that it **should have been written to** return me a `Boolean`?

Reply:
I'm not saying any of those things in particular. This is me wondering why that returns a string. :)

Reply:
Ok, so I've refreshed my memory on properties. You're right that a _lot_ of properties are strings. There are also quite a few which are `Boolean`, though, e.g., `SELECTED`, `OBSCURED_TO_ME`, `OBSCURED_TO_OTHERS`, `NO_STACK`.

So you could have `MatCargo.getProperty()` (and `getLocalizedProperty()`) return `true` instead of `"true"` and change the test on the value here (and in the other three places where `IS_CARGO` is used) to `Boolean.TRUE.equals()`.

Reply:
Gotcha - all fixed